oomagent/test/test_channel_join.sh

```diff
@@ -104,9 +104,43 @@ EOF
   assert_eq "$case" "$expected" "$actual"
 }
 
+no_entity_rows() {
+  case="no entity rows"
+  arg=$(cat <<-EOF
+{
+  "join_features": [
+    "driver_stats.conv_rate"
+  ]
+}
+EOF
+)
+  actual=$(testgrpc ChannelJoin <<<"$arg" 2>&1 || flag=true)
+  expected=$(cat <<-EOF
+EOF
+)
+  assert_eq "$case" "$expected" "$actual"
+}
+
+no_data() {
+   case="no data"
+  arg=$(cat <<-EOF
+EOF
+)
+  actual=$(testgrpc ChannelJoin <<<"$arg" 2>&1 || flag=true)
+  expected=$(cat <<-EOF
+ERROR:
+  Code: Internal
+  Message: invalid request: empty feature
+EOF
+)
+  assert_eq "$case" "$expected" "$actual"
+}
+
 main() {
   successful_cases
   failed_cases
+  no_entity_rows
+  no_data
```

Comment:
you could wrap `no_data` and `no_entity_rows` to `failed_cases`

Reply:
done